bitmark-references/bitmark-node-software/mining.md

```diff
@@ -0,0 +1,117 @@
+
+```
+printf '%s' 'hello world' | argon2 'hello world' -d -l 32 -m 17 -t 4 -p 1 -r | awk '{for(i=length($1);i>0;i-=2)x=x substr($1,i-1,2);print x}'
+
+f8a17bc25cb53e848e2d09811ade4b8a037f628443661b88611faf5d9a5a1f33
+```
+
+A block hash contains following data:
+
+1. block record version
+
+    blockchain header version
+
+1. transacion count
+
+    number of transactions in the block
+
+1. merkle root
+
+    the hash of all transaction hashes inside the block, if verified transactions are less than block allowable limit (9999), then all verified transactions will be put into a block. If verified transactions above block allowable limit, random transactions will be selected
+
+1. timestamp
+
+    block generated time
+
+1. difficulty
+
+    hashing difficulty
+
+1. nonce
+
+    random string to make hash fits difficuty
+
+## Difficulty
+
+Difficulty is a number to decide which hash is valid, higher difficulty means harder to find a valid hash. A hash meets difficulty means hash vaue is less than or equal to difficulty value.
+
+The hash is considered as a 256 bit fixed point value composeda as following two parts:
+
+    [8 bits] [248 bits]
+    exponent mantissa
+
+Difficulty value is encoded by 64 bits, including 8 bits of exponent and 56 bits of mantissa.
+
+    [56 bits] [8 bits]
+    mantissa  exponent = 64 bits unsigned value
+
+Mantissa is normalized so that most significat bits is one and can be droppped leaving 56 bits to store, itt is right shifted by exponent +8 from the most significant bit.
+
+The difficulty `One` is defined as `00 ff  ff ff  ff ff  ff ff`, represents 256 bits value `0ff ffff ffff ffff 8000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000`.
+
+The `Two` is doubled of difficulty `One`: `01ffffffffffffff`, represents 256 bits value `007fffffffffffffc00000000000000000000000000000000000000000000000`.
+
+## Communication
+
+`bitmarkd` sends hashing job to `recorderd`. Each `bitmarkd` posses a job queue to keep records of what job has been sent to `recorderd`. If `recorderd` finds possible hash that might fits difficulty, it sends result back to `bitmarkd`, and `bitmarkd` returns verified result back to `recorderd`.
+
+The message format from `recorderd` to `bitmarkd` as follows:
+
+1. request
+
+    a string "block.nonce" to represent what kind of request
+
+1. job
+
+    `bitmarkd` provided number to denote job in queue
+
+1. packed
+
+    nonce that may makes hash meets difficulty
+
+The message from `bitmarkd` to reply `recorderd` as follows:
+
+1. job
+
+    job number
+
+2. ok
+
+    denote hash is valid or not
+
+## Block Verification and Broadcast
+
+If a hash is valid, then `bitmarkd` stores that block locally. While sotring block, it will check block data again, including correctness of hash, chain is linked to previous one, payment informatino is valid, etc. After block is successfully stored, `bitmarkd` will broadcast block information to all connected nodes.
```

Comment:
Please run a misspelling checker and also check how I made the other suggestions.